src/tools/rbd_mirror/image_replayer/snapshot/Replayer.cc

```diff
@@ -272,6 +277,26 @@ bool Replayer<I>::get_replay_status(std::string* description,
         static_cast<float>(std::max<uint64_t>(1U, m_local_object_count)));
   }
 
+  m_bytes_per_second(0);
+  auto bytes_per_second = m_bytes_per_second.get_average();
+  root_obj["bytes_per_second"] = round_to_two_places(bytes_per_second);
+
+  auto bytes_per_snapshot = boost::accumulators::rolling_mean(
+    m_bytes_per_snapshot);
+  root_obj["bytes_per_snapshot"] = round_to_two_places(bytes_per_snapshot);
+
+  auto pending_bytes = bytes_per_snapshot * m_pending_snapshots;
+  if (bytes_per_second > 0 && m_pending_snapshots > 0) {
+    auto seconds_until_synced = round_to_two_places(
+      pending_bytes / bytes_per_second);
```

Comment:
the same here: in this estimate we don't include new bytes that are going to be added to the image during sync. If we had something like `new_snapshots_per_second`, it could be:
```
seconds_until_synced = bytes_per_snapshot * pending_snapshots / bytes_per_second * (1 + new_snapshots_per_second * bytes_per_snapshot / bytes_per_second)
```

Reply:
We we can measure new snapshots in "snapshots per second", I think we have a problem. :wink: 

Once a new snapshot is added, the stats will update after it rescans the image since we now track how many snapshots we need to sync + the average number of bytes we copy per snapshot.

Reply:
Sure, "snapshots per second" is expected to be a float less than 1, e.g. if snapshots are created hourly it would be `1 / 3600`. Anyway, thinking more about it, I tend to agree there is no need in overcomplicating this.